src/core/context-management/ContextManager.ts

```diff
@@ -380,70 +401,174 @@ export class ContextManager {
 	}
 
 	/**
-	 * if there is any truncation, and there is no other alteration already set, alter the assistant message to indicate this occurred
+	 * if there is any truncation and there is no other alteration already set, alter the assistant message to indicate this occurred
 	 */
 	private applyStandardContextTruncationNoticeChange(timestamp: number): boolean {
 		if (!this.contextHistoryUpdates.has(1)) {
 			// first assistant message always at index 1
 			const innerMap = new Map<number, ContextUpdate[]>()
-			innerMap.set(0, [[timestamp, "text", [formatResponse.contextTruncationNotice()]]]) // alter message text at index 0
-			this.contextHistoryUpdates.set(1, innerMap)
+			innerMap.set(0, [[timestamp, "text", [formatResponse.contextTruncationNotice()], []]])
+			this.contextHistoryUpdates.set(1, [0, innerMap]) // EditType is undefined for first assistant message
```

Comment:
It looks like "EditType" is the type we should use above based on the comment here.